tests/components/kostal_plenticore/test_select.py

```diff
@@ -0,0 +1,64 @@
+"""Test the Kostal Plenticore Solar Inverter select platform."""
+from unittest.mock import Mock
+
+from kostal.plenticore import PlenticoreApiClient, SettingsData
+
+from homeassistant.components.kostal_plenticore.const import DOMAIN
+from homeassistant.components.kostal_plenticore.helper import Plenticore
+from homeassistant.components.kostal_plenticore.select import async_setup_entry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity import DeviceInfo
+
+from tests.common import MockConfigEntry
+
+
+async def test_select(hass: HomeAssistant, mock_config_entry: MockConfigEntry):
+    """Test that select entities are added."""
+
+    plenticore: Plenticore = Mock(spec=Plenticore)
+    hass.data[DOMAIN] = {mock_config_entry.entry_id: plenticore}
```

Comment:
Patch plenticore instead and set up the integration by setting up the mock config entry.

https://developers.home-assistant.io/docs/development_testing#writing-tests-for-integrations